File: sdk/r/R/contacts.R
#' Contacts: manage account contacts
#' @param http HttpClient object
#' @return Resource object
#' @export
ContactsResource <- function(http) {
  env <- new.env(parent = emptyenv())
  env$http <- http
  class(env) <- "ContactsResource"
  env
}

#' GET /api/contact - Return a list of contacts on this account
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
list <- function(self, ...) {
  self$http$get("/api/contact", list(...))
}

#' POST /api/contact - Create new contact account
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
create <- function(self, ...) {
  self$http$post("/api/contact", list(...))
}

#' GET /api/contact/privileges - List possible contact privileges
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getPrivileges <- function(self, ...) {
  self$http$get("/api/contact/privileges", list(...))
}

#' GET /api/contact/{id} - Get contact details
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
get <- function(self, ...) {
  self$http$get("/api/contact/{id}", list(...))
}

#' PUT /api/contact/{id} - Update contact details
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
update <- function(self, ...) {
  self$http$put("/api/contact/{id}", list(...))
}

File: sdk/c/src/contacts.c
#include "hitechcloud/contacts.h"
#include <stdlib.h>


/* GET /api/contact - Return a list of contacts on this account */
char *hc_contacts_list(hc_http_client_t *client) {
    return hc_get(client, "/api/contact", NULL);
}

/* POST /api/contact - Create new contact account */
char *hc_contacts_create(hc_http_client_t *client) {
    return hc_post(client, "/api/contact", NULL);
}

/* GET /api/contact/privileges - List possible contact privileges */
char *hc_contacts_get_privileges(hc_http_client_t *client) {
    return hc_get(client, "/api/contact/privileges", NULL);
}

/* GET /api/contact/{id} - Get contact details */
char *hc_contacts_get(hc_http_client_t *client) {
    return hc_get(client, "/api/contact/{id}", NULL);
}

/* PUT /api/contact/{id} - Update contact details */
char *hc_contacts_update(hc_http_client_t *client) {
    return hc_put(client, "/api/contact/{id}", NULL);
}

